mono/metadata/object.c

```diff
@@ -1932,23 +1930,22 @@ mono_class_create_runtime_vtable (MonoDomain *domain, MonoClass *klass, MonoErro
 	vt->domain = domain;
 
 	mono_class_compute_gc_descriptor (klass);
-		/*
-		 * We can't use typed allocation in the non-root domains, since the
-		 * collector needs the GC descriptor stored in the vtable even after
-		 * the mempool containing the vtable is destroyed when the domain is
-		 * unloaded. An alternative might be to allocate vtables in the GC
-		 * heap, but this does not seem to work (it leads to crashes inside
-		 * libgc). If that approach is tried, two gc descriptors need to be
-		 * allocated for each class: one for the root domain, and one for all
-		 * other domains. The second descriptor should contain a bit for the
-		 * vtable field in MonoObject, since we can no longer assume the 
-		 * vtable is reachable by other roots after the appdomain is unloaded.
-		 */
-#ifdef HAVE_BOEHM_GC
-	if (domain != mono_get_root_domain () && !mono_dont_free_domains)
+	/*
+	 * For Boehm:
+	 * We can't use typed allocation in the non-root domains, since the
+	 * collector needs the GC descriptor stored in the vtable even after
+	 * the mempool containing the vtable is destroyed when the domain is
+	 * unloaded. An alternative might be to allocate vtables in the GC
+	 * heap, but this does not seem to work (it leads to crashes inside
+	 * libgc). If that approach is tried, two gc descriptors need to be
+	 * allocated for each class: one for the root domain, and one for all
+	 * other domains. The second descriptor should contain a bit for the
+	 * vtable field in MonoObject, since we can no longer assume the
+	 * vtable is reachable by other roots after the appdomain is unloaded.
+	 */
+	if (!mono_gc_is_moving () && domain != mono_get_root_domain () && !mono_dont_free_domains)
```

Comment:
Probably fine, but this really has nothing to do with moving GC. Just GC that needs object descriptors valid after domain unload. 

I see we are just using `mono_gc_is_moving` as proxy for SGen so maybe that's okay.